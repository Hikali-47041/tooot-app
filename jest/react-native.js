jest.mock('react-native/Libraries/LayoutAnimation/LayoutAnimation', () => ({
  ...require.requireActual(
    'react-native/Libraries/LayoutAnimation/LayoutAnimation'
  ),
  configureNext: jest.fn()
}))
