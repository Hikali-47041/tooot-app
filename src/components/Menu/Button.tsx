import React from 'react'
import {
  Alert,
  AlertButton,
  AlertOptions,
  Pressable,
  StyleSheet,
  Text,
  View
} from 'react-native'
import { useTheme } from '@utils/styles/ThemeManager'
import { StyleConstants } from '@utils/styles/constants'

export interface Props {
  text: string
  destructive?: boolean
  alertOption?: {
    title: string
    message?: string | undefined
    buttons?: AlertButton[] | undefined
    options?: AlertOptions | undefined
  }
}

const Core: React.FC<Props> = ({ text, destructive = false }) => {
  const { theme } = useTheme()

  return (
    <View style={styles.core}>
      <Text style={{ color: destructive ? theme.error : theme.primary }}>
        {text}
      </Text>
    </View>
  )
}

const MenuButton: React.FC<Props> = ({ ...props }) => {
  const { theme } = useTheme()

  return (
    <Pressable
      style={[styles.base, { borderBottomColor: theme.separator }]}
      onPress={() =>
        props.alertOption &&
        Alert.alert(
          props.alertOption.title,
          props.alertOption.message,
          props.alertOption.buttons,
          props.alertOption.options
        )
      }
    >
      <Core {...props} />
    </Pressable>
  )
}

const styles = StyleSheet.create({
  base: {
    height: 50,
    borderBottomWidth: 1
  },
  core: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    paddingLeft: StyleConstants.Spacing.Global.PagePadding,
    paddingRight: StyleConstants.Spacing.Global.PagePadding
  }
})

export default MenuButton
