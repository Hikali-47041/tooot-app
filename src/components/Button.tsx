import React from 'react'
import { Pressable, StyleSheet, Text } from 'react-native'
import { StyleConstants } from 'src/utils/styles/constants'
import { useTheme } from 'src/utils/styles/ThemeManager'

export interface Props {
  onPress: () => void
  text: string
  fontSize?: 'S' | 'M' | 'L'
}

const Button: React.FC<Props> = ({ onPress, text, fontSize = 'M' }) => {
  const { theme } = useTheme()

  return (
    <Pressable
      onPress={onPress}
      style={[styles.button, { borderColor: theme.primary }]}
    >
      <Text
        style={[
          styles.text,
          { color: theme.primary, fontSize: StyleConstants.Font.Size[fontSize] }
        ]}
      >
        {text}
      </Text>
    </Pressable>
  )
}

const styles = StyleSheet.create({
  button: {
    paddingTop: StyleConstants.Spacing.S,
    paddingBottom: StyleConstants.Spacing.S,
    paddingLeft: StyleConstants.Spacing.M,
    paddingRight: StyleConstants.Spacing.M,
    borderWidth: 1,
    borderRadius: 100
  },
  text: {
    textAlign: 'center'
  }
})

export default Button
