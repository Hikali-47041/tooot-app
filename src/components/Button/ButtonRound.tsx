import { Feather } from '@expo/vector-icons'
import React from 'react'
import { Pressable, StyleSheet } from 'react-native'
import { StyleConstants } from '@utils/styles/constants'
import { useTheme } from '@utils/styles/ThemeManager'

export interface Props {
  styles: any
  onPress: () => void
  icon: any
  size?: 'S' | 'M' | 'L'
  coordinate?: 'center' | 'default'
}

const ButtomRound: React.FC<Props> = ({
  styles: extraStyles,
  onPress,
  icon,
  size = 'M',
  coordinate = 'default'
}) => {
  const { theme } = useTheme()
  const dimension =
    StyleConstants.Spacing.S * 1.5 + StyleConstants.Font.Size[size]

  return (
    <Pressable
      style={[
        styles.base,
        extraStyles,
        {
          backgroundColor: theme.backgroundOverlay,
          ...(coordinate === 'center' && {
            transform: [
              { translateX: -dimension / 2 },
              { translateY: -dimension / 2 }
            ]
          })
        }
      ]}
      onPress={onPress}
    >
      <Feather
        name={icon}
        size={StyleConstants.Font.Size[size]}
        color={theme.primaryOverlay}
      />
    </Pressable>
  )
}

const styles = StyleSheet.create({
  base: {
    position: 'absolute',
    padding: StyleConstants.Spacing.S * 1.5,
    borderRadius: StyleConstants.Spacing.XL
  }
})

export default ButtomRound
