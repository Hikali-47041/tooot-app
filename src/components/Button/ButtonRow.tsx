import { Feather } from '@expo/vector-icons'
import React from 'react'
import { Pressable, StyleProp, StyleSheet, Text, ViewStyle } from 'react-native'
import { StyleConstants } from '@utils/styles/constants'
import { useTheme } from '@utils/styles/ThemeManager'

type PropsBase = {
  onPress: () => void
  disabled?: boolean
  buttonSize?: 'S' | 'M'
  size?: 'S' | 'M' | 'L'
  style?: StyleProp<ViewStyle>
}

export interface PropsText extends PropsBase {
  text: string
  icon?: any
}

export interface PropsIcon extends PropsBase {
  text?: string
  icon: any
}

const ButtonRow: React.FC<PropsText | PropsIcon> = ({
  onPress,
  disabled = false,
  buttonSize = 'M',
  text,
  icon,
  size = 'M',
  style: customStyle
}) => {
  const { theme } = useTheme()

  return (
    <Pressable
      {...(!disabled && { onPress })}
      style={[
        customStyle,
        styles.button,
        {
          paddingLeft:
            StyleConstants.Spacing.M -
            (icon ? StyleConstants.Font.Size[size] / 2 : 0),
          paddingRight:
            StyleConstants.Spacing.M -
            (icon ? StyleConstants.Font.Size[size] / 2 : 0),
          borderColor: disabled ? theme.secondary : theme.primary,
          paddingTop: StyleConstants.Spacing[buttonSize === 'M' ? 'S' : 'XS'],
          paddingBottom: StyleConstants.Spacing[buttonSize === 'M' ? 'S' : 'XS']
        }
      ]}
    >
      {icon ? (
        <Feather
          name={icon}
          size={StyleConstants.Font.Size[size]}
          color={disabled ? theme.secondary : theme.primary}
        />
      ) : (
        <Text
          style={[
            styles.text,
            {
              color: disabled ? theme.secondary : theme.primary,
              fontSize: StyleConstants.Font.Size[size]
            }
          ]}
        >
          {text}
        </Text>
      )}
    </Pressable>
  )
}

const styles = StyleSheet.create({
  button: {
    borderWidth: 1.25,
    borderRadius: 100,
    alignItems: 'center'
  },
  text: {
    textAlign: 'center'
  }
})

export default ButtonRow
