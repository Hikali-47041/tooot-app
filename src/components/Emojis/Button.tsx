import Icon from '@components/Icon'
import { StyleConstants } from '@utils/styles/constants'
import { useTheme } from '@utils/styles/ThemeManager'
import React, { useContext } from 'react'
import { Keyboard, Pressable } from 'react-native'
import EmojisContext from './helpers/EmojisContext'

const EmojisButton: React.FC = () => {
  const { colors } = useTheme()
  const { emojisState, emojisDispatch } = useContext(EmojisContext)

  return (
    <Pressable
      disabled={!emojisState.emojis || !emojisState.emojis.length}
      onPress={() => {
        const targetProps = emojisState.inputProps?.find(props => props.ref.current?.isFocused())
        if (!targetProps) {
          return
        }
        if (emojisState.targetProps === null) {
          Keyboard.dismiss()
        }
        emojisDispatch({ type: 'target', payload: targetProps })
      }}
      hitSlop={StyleConstants.Spacing.S}
      style={{ alignSelf: 'flex-end', padding: StyleConstants.Spacing.Global.PagePadding }}
      children={
        <Icon
          name={emojisState.emojis && emojisState.emojis.length ? 'Smile' : 'Meh'}
          size={24}
          color={
            emojisState.emojis && emojisState.emojis.length
              ? colors.primaryDefault
              : colors.disabled
          }
        />
      }
    />
  )
}

export default EmojisButton
