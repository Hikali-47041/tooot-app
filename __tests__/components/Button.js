import React from 'react'
import {
  toBeDisabled,
  toHaveStyle,
  toHaveTextContent
} from '@testing-library/jest-native'
import { cleanup, fireEvent, render } from '@testing-library/react-native/pure'

import Button from '@components/Button'

expect.extend({ toBeDisabled, toHaveStyle, toHaveTextContent })

describe('Testing component button', () => {
  afterEach(cleanup)

  describe('static button', () => {
    it('with text only', () => {
      const onPress = jest.fn()
      const { getByTestId, toJSON } = render(
        <Button type='text' content='Test Button' onPress={onPress} />
      )
      fireEvent.press(getByTestId('base'))

      expect(onPress).toHaveBeenCalled()
      expect(onPress).toHaveBeenCalledTimes(1)
      expect(getByTestId('text')).toHaveTextContent('Test Button')
      expect(toJSON()).toMatchSnapshot()
    })

    it('with icon only', () => {
      const onPress = jest.fn()
      const { getByTestId, toJSON } = render(
        <Button type='icon' content='X' onPress={onPress} />
      )
      fireEvent.press(getByTestId('base'))

      expect(onPress).toHaveBeenCalled()
      expect(onPress).toHaveBeenCalledTimes(1)
      expect(toJSON()).toMatchSnapshot()
    })

    it('loading state', () => {
      const { getByTestId, toJSON } = render(
        <Button type='text' content='test' onPress={jest.fn()} loading />
      )

      expect(getByTestId('base')).toBeDisabled()
      expect(toJSON()).toMatchSnapshot()
    })

    it('disabled state', () => {
      const { getByTestId, toJSON } = render(
        <Button type='text' content='test' onPress={jest.fn()} disabled />
      )

      expect(getByTestId('base')).toBeDisabled()
      expect(toJSON()).toMatchSnapshot()
    })

    it('apply custom styling', () => {
      const { getByTestId, toJSON } = render(
        <Button
          type='text'
          content='test'
          onPress={jest.fn()}
          style={{ backgroundColor: 'black' }}
        />
      )

      expect(getByTestId('base')).toHaveStyle({ backgroundColor: 'black' })
      expect(toJSON()).toMatchSnapshot()
    })
  })

  describe('dynamic button', () => {
    it('from default to loading', () => {
      const onPress = jest.fn()
      const { getByTestId, rerender } = render(
        <Button type='text' content='test' onPress={onPress} />
      )
      rerender(<Button type='text' content='test' onPress={onPress} loading />)

      expect(getByTestId('base')).toBeDisabled()
    })

    it('from default to disabled', () => {
      const onPress = jest.fn()
      const { getByTestId, rerender } = render(
        <Button type='text' content='test' onPress={onPress} />
      )
      rerender(<Button type='text' content='test' onPress={onPress} disabled />)

      expect(getByTestId('base')).toBeDisabled()
    })
  })
})
