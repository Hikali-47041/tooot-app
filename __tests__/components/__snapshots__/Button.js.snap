// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component button static button apply custom styling 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 100,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "rgb(250, 250, 250)",
        "borderColor": "rgb(18, 18, 18)",
        "borderWidth": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
      Object {
        "backgroundColor": "black",
      },
    ]
  }
  testID="base"
>
  <Text
    style={
      Object {
        "color": "rgb(18, 18, 18)",
        "fontSize": 16,
        "fontWeight": undefined,
        "opacity": 1,
      }
    }
    testID="text"
  >
    test
  </Text>
</View>
`;

exports[`Testing component button static button disabled state 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 100,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "rgb(250, 250, 250)",
        "borderColor": "rgb(135, 135, 135)",
        "borderWidth": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
      undefined,
    ]
  }
  testID="base"
>
  <Text
    style={
      Object {
        "color": "rgb(135, 135, 135)",
        "fontSize": 16,
        "fontWeight": undefined,
        "opacity": 1,
      }
    }
    testID="text"
  >
    test
  </Text>
</View>
`;

exports[`Testing component button static button loading state 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 100,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "rgb(250, 250, 250)",
        "borderColor": "rgb(135, 135, 135)",
        "borderWidth": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
      undefined,
    ]
  }
  testID="base"
>
  <Text
    style={
      Object {
        "color": "rgb(18, 18, 18)",
        "fontSize": 16,
        "fontWeight": undefined,
        "opacity": 0,
      }
    }
    testID="text"
  >
    test
  </Text>
  <View
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 16,
          "justifyContent": "center",
          "opacity": 1,
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
          "width": 16,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgb(135, 135, 135)",
            "borderRadius": 2,
            "height": 4,
            "position": "absolute",
            "transform": Array [
              Object {
                "rotate": "73.27536734311887deg",
              },
              Object {
                "translateY": -6,
              },
              Object {
                "scale": 0.7,
              },
            ],
            "width": 4,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "rgb(135, 135, 135)",
            "borderRadius": 2,
            "height": 4,
            "position": "absolute",
            "transform": Array [
              Object {
                "rotate": "46.49829517703514deg",
              },
              Object {
                "translateY": -6,
              },
              Object {
                "scale": 0.8008696779414123,
              },
            ],
            "width": 4,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "rgb(135, 135, 135)",
            "borderRadius": 2,
            "height": 4,
            "position": "absolute",
            "transform": Array [
              Object {
                "rotate": "25.743213498935145deg",
              },
              Object {
                "translateY": -6,
              },
              Object {
                "scale": 0.8875624559768125,
              },
            ],
            "width": 4,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "rgb(135, 135, 135)",
            "borderRadius": 2,
            "height": 4,
            "position": "absolute",
            "transform": Array [
              Object {
                "rotate": "11.201058030774364deg",
              },
              Object {
                "translateY": -6,
              },
              Object {
                "scale": 0.9510040862404615,
              },
            ],
            "width": 4,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "rgb(135, 135, 135)",
            "borderRadius": 2,
            "height": 4,
            "position": "absolute",
            "transform": Array [
              Object {
                "rotate": "2.731234791722257deg",
              },
              Object {
                "translateY": -6,
              },
              Object {
                "scale": 0.9881665278710133,
              },
            ],
            "width": 4,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "rgb(135, 135, 135)",
            "borderRadius": 2,
            "height": 4,
            "position": "absolute",
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
              Object {
                "translateY": -6,
              },
              Object {
                "scale": 1,
              },
            ],
            "width": 4,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Testing component button static button with icon only 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 100,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "rgb(250, 250, 250)",
        "borderColor": "rgb(18, 18, 18)",
        "borderWidth": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
      undefined,
    ]
  }
  testID="base"
>
  <Text />
</View>
`;

exports[`Testing component button static button with text only 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 100,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "rgb(250, 250, 250)",
        "borderColor": "rgb(18, 18, 18)",
        "borderWidth": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
      undefined,
    ]
  }
  testID="base"
>
  <Text
    style={
      Object {
        "color": "rgb(18, 18, 18)",
        "fontSize": 16,
        "fontWeight": undefined,
        "opacity": 1,
      }
    }
    testID="text"
  >
    Test Button
  </Text>
</View>
`;
